autoscaling/cluster-resource-overrides.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Cluster-Resource-Overrides-Enablement
+authors:
+  - "@deads2k"
+reviewers:
+  - "@sttts"
+  - "@derekwaynecarr"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-09-11
+last-updated: 2019-09-11
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# ClusterResourceOverrides Enablement
+
+The `autoscaling.openshift.io/ClusterResourceOverride` cannot be enabled in 4.x.  The plugin already exists, this design 
+is about how we make it possible for a customer to enable the feature.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The `autoscaling.openshift.io/ClusterResourceOverride` admission plugin is an uncommonly used admission plugin with configuration
+values.  Because it is uncommonly used, it doesn't fit well with our targeted configuration which aims to avoid adding
+lots of intricately documented knobs.  Instead of wiring the admission plugin via a kube-apiserver operator, we can create
+a mutating admission webhook based on the [generic-admission-server](https://github.com/openshift/generic-admission-server)
+and install it via OLM.
+
+## Motivation
+
+The `autoscaling.openshift.io/ClusterResourceOverride` admission plugin is used for over-commit, let's stipulate that it is
+important enough to enable.  The kube-apiserver is designed to be extended using mutating admission webhooks, we have the
+technology to easily build one, and we have the ability to create a simple operator to manage it.  We want to enable the 
+feature using a pattern that we can extend to other admission plugins that can scale beyond the small team that maintains
+the kube-apiserver.
+
+### Goals
+
+1. Enable the `autoscaling.openshift.io/ClusterResourceOverride` admission plugin that is used for overcommit.
+2. Use existing extension points, libraries, and installation mechanisms in the manner we would recommend to 
+ external teams.
+3. Have a fairly straightforward way to install and enable this admission plugin.
+4. Rebootstrapping must be possible.
+
+### Non-Goals
+
+1. Revisit how `autoscaling.openshift.io/ClusterResourceOverride` works.  We're lifting it as-is.
+2. Couple a slow moving admission plugin to a fast moving kube-apiserver.
+
+### Open Questions
+
+1. Do we need to protect openshift resources from being overcommitted?  Perhaps the cluster-admin's intent is exactly that.
+2. We cannot uniformly apply protection just to our payload resources, how do we position this?
```

Comment:
we could have an opt-in per namespace behavior.  the world has evolved around us since this original plug-in was conceived, it is possible that operators that take resource requirements as input could attempt to reconcile a overriden resource, but that is not much different than limit ranges already.